keps/1834-copy-labels-into-workload/README.md

```diff
@@ -0,0 +1,103 @@
+# KEP-1834: Copy some labels from Pods/Jobs into the Workload object 
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+When creating a kueue workload, copy the job (or pod) labels into the workload object (see https://github.com/kubernetes-sigs/kueue/issues/1834). Allow to configure which labels should be copied.
+
+## Motivation
+
+Currently the workloads do not "inherit" any of the labels of the jobs based on which they are created. Since workloads in Kueue are internal representations of the Kubernetes jobs, allowing to easily copy the labels of the job object into the workload object is natural and useful. For instance it would facilitate selection/filtering of the workloads by the administrators. 
+
+### Goals
+
+* Establish a mechanism to configure which set of labels should be copied.
+* Copy the selected labels when creating the workload based on a job.
+
+### Non-Goals
+This proposal does not contain any form of validation or analysis of the labels as they are going to be copied from one object to another.
+
+## Proposal
+
+We want to do the following API change. The proposal is to add a field named `labelKeysToCopy` into the configuration API (under `Integrations`). This field will hold a list of keys of labels that should be copied. This configuration will be global in the sense that it will apply to all the job frameworks. Since the list `labelKeysToCopy` will be empty by default, this change will not affect the existing functionality. 
+
+We will not require for all the labels with keys from `labelKeysToCopy` to be present at the job object. When some of the labels will not be present at the job object, this label will not be assigned to the created workload object.
+
+A case that requires more attention is creating workloads from pod groups because in that case a single workload is based on multiple pods (each of which might have labels). We propose:
+ * When a label from the `labelKeysToCopy` list will be present at some of the pods from the group and the value of this label on all these pods will be identical then this label will be copied into the workload. Note that we do not require that all the pods have the label but all that do have must have the same value.
+ * When multiple pods from the group will have the label with different values, we will raise an exception.
```

Comment:
Is the Workload object created only when we have all the pods in a group created? If it is created when the first pod is created, then we may not have a chance to raise the exception. 
Also, related:
- when is the exception raised, it is during workload creation, or later? If later, how is the error communicated to the user?
- what happens when the workload already exists, and the user creates a new pod with a non-matching value?

Reply:
> As far as I know the workload object is created when all the pods are created.

This would simplify this indeed. Could you please double check on a running system?

> The exception is raised during workload creation.
> The new feature is only applied during the creation of a new workload so if the user creates another pod later it will have no effect on the labels on the running workload.

sgtm, please add a sentence along the lines to clarify in KEP for future readers.

Reply:
> As far as I know the workload object is created when all the pods are created.
That is correct.

Reply:
Done. Added a sentence to KEP clarifying this.

Reply:
Thanks for clarifying and the update.